runtime/codert_vm/decomp.cpp

```diff
@@ -1408,7 +1408,20 @@ jitDataBreakpointAdded(J9VMThread * currentThread)
 			/* Make all future compilations inline the field watch code */
 			jitConfig->inlineFieldWatches = TRUE;
 		} else {
+			/* jitClassesRedefined expects the caller to hold the class unload mutex exclusively. */
+#if defined(J9VM_JIT_CLASS_UNLOAD_RWMONITOR)
+			omrthread_rwmutex_enter_write(vm->classUnloadMutex);
```

Comment:
`jitDataBreakpointAdded` is invoked from only one location (`jvmtiWatchedField.c::setFieldWatch`) which acquires exclusive VM access. No other thread should run with exclusive VM access. Do we need to acquire `classUnloadMutex` here?

Reply:
As described in my reply to the other thread, compilations can and usually do run without VM access (for most of the time). Additionally, much of `jitClassesRedefined()` expects to exclusively hold the class unload mutex (as stated in the comment just above this line). AFAICT the class unload mutex can at least sometimes be crucial to establishing memory ordering between `setAllCompilationsShouldBeInterrupted()` and `CompilationInfoPerThreadBase::compilationShouldBeInterrupted()`. It's not obvious to me whether or not there are other ways in which `jitClassesRedefined()` relies on the fact that it holds the class unload mutex.

That said, I'm not sure why we need to call `jitClassesRedefined()` here in the first place. If we're in fast HCR mode, then it won't do anything. But unless we can establish firmly that it's unneeded, I'd prefer to err on the side of leaving the call in place, in which case the mutex should be held.